src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/ProfilerControllerTest.php

```diff
@@ -135,4 +124,28 @@ public function testSearchResult()
         $response = $controller->searchResultsAction($request, 'empty');
         $this->assertEquals(200, $response->getStatusCode());
     }
+
+    public function provideController()
+    {
+        $urlGenerator1 = $this->getMock('Symfony\Component\Routing\Generator\UrlGeneratorInterface');
+        $twig1 = $this->getMockBuilder('Twig_Environment')->disableOriginalConstructor()->getMock();
+        $profiler1 = $this
+            ->getMockBuilder('Symfony\Component\HttpKernel\Profiler\Profiler')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $urlGenerator2 = $this->getMock('Symfony\Component\Routing\Generator\UrlGeneratorInterface');
+        $twig2 = $this->getMockBuilder('Twig_Environment')->disableOriginalConstructor()->getMock();
```

Comment:
I don't think you should build mocks in the data provider. This can lead to weird things as these mocks are created way earlier than the test being run, and with a different instance of the TestCase (I don't remember how it works for PHPUnit_MockObject, but I'm 100% sure it does not work properly when using Prophecy as expectations won't be checked for instance).

I suggest you to make a data provider only about getting a boolean whether CSP should be enabled or no